index.html

```diff
@@ -204,7 +304,73 @@
 						bindKey: { win: 'Ctrl-Enter', mac: 'Command-Enter' },
 						exec: vm.run_code,
 					})
+
+					// Control-S -> Save file
+					vm.editor.commands.addCommand({
+						name: 'save_code',
+						bindKey: { win: 'Ctrl-S', mac: 'Ctrl-S' },
+						exec: vm.save_code,
+					})
+				},
+
+				save_code() {
+					const vm = this
+
+					// Get script name
+					var script_name = prompt("Enter script name:")
+
+					// If user hits the cancel button
+					if (script_name == null){
+						return
+					}
+
+					if (script_name.length > 0){
+	
+						// Save Vue script name data with prompt input. might be useful
+						vm.script_name = script_name
+						document.title = script_name
+
+						cookie = {
+							lang: vm.selected_lang, 
+							script_name: script_name, 
+							display: script_name,
+							code: vm.editor.getValue()}
+						cookie_str = JSON.stringify(cookie)
+
+						console.log("Saving cookie:", cookie_str)
+
+						// Save cookie to browser
+						createCookie(script_name, cookie_str, 2*420)  // save cookie to browser
+
+						// Save cookie to Vue
+						this.cookies.push(cookie)
```

Comment:
use `vm.*` instead of `this.*`.
Makes it less bug prone to move in & out of lambdas.

Reply:
updated with fix